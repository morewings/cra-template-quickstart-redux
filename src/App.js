import React from 'react';

// components
import Sample from 'components/Sample/Sample.container';
import SampleWrapped from 'components/SampleWrapped';

const App = props => (
  <div className="App">
    <Sample />
    <SampleWrapped />
  </div>
);

export default App;
