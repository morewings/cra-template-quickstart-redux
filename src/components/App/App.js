import React from 'react';
import Counter from '../Counter';
import Random from '../Random';
import classes from './App.module.css';

const App = () => (
  <div className={classes.container}>
    <Counter />
    <Random />
  </div>
);

export default App;
