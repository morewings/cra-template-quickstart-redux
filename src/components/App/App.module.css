.container {
  display: flex;
  margin: 0 auto;
  width: 960px;
}
