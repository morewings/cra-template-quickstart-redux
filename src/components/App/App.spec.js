import React from 'react';
import {Provider} from 'react-redux';
import configureStore from 'redux-mock-store';
import {render} from '@testing-library/react';
import App from './App';

describe('components > App', () => {
  const mockStore = configureStore([]);
  const store = mockStore({
    count: {
      value: 6,
    },
    random: {
      isLoading: false,
      hasError: false,
      isFulfilled: false,
    },
  });

  it('renders without crashing', () => {
    /**
     * `asFragment`:
     * @see https://testing-library.com/docs/react-testing-library/api#asfragment
     * `wrapper`
     * @see https://testing-library.com/docs/react-testing-library/api#wrapper
     */
    const {asFragment} = render(<App />, {
      wrapper: ({children}) => <Provider store={store}>{children}</Provider>,
    });

    /**
     * Basic snapshot test to make sure, that rendered component
     * matches expected footprint.
     */
    expect(asFragment()).toMatchSnapshot();
  });
});
