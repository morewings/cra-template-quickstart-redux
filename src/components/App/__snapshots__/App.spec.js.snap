// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components > App renders without crashing 1`] = `
<DocumentFragment>
  <div
    class="container"
  >
    <div
      class="counter"
    >
      <h2
        class="header"
      >
        Sync counter
      </h2>
      <button
        class="button"
        type="button"
      >
        Increment by one
      </button>
      <div>
        Total value: 
        <strong>
          6
        </strong>
      </div>
    </div>
    <div
      class="counter"
    >
      <h2
        class="header"
      >
        Async Random
      </h2>
      <button
        class="button"
        type="button"
      >
        Get random number
      </button>
      <div>
        Click the button to get random number
      </div>
    </div>
  </div>
</DocumentFragment>
`;
